stdlib: Outline integer parsing code in FixedWidthInteger.init(_: radix:) on the slow paths

aschwaighofer opened the pull request:
This reduces code size by 20k on an app that uses this function.

rdar://32519912

aschwaighofer committed:
stdlib: Outline integer parsing code in FixedWidthInteger.init(_: radix:) on the slow paths

This reduces code size by 20k on an app that uses this function.

rdar://32519912

aschwaighofer commented:
@swift-ci Please benchmark

dabrahams commented on stdlib/public/core/IntegerParsing.swift:
```diff
@@ -83,6 +83,16 @@ where CodeUnits.Element : UnsignedInteger {
   }
 }
 
+@inline(never)
```
Why isn't this an extension method on `FixedWidthInteger`?  As an implementation detail of that `init?` it rather makes sense to put it right next to it in the type.

aschwaighofer replied:
I followed your example of the none ...Outlined version next to it?

aschwaighofer committed:
stdlib: Move _parseASCIIOutlined into the FixedWidthInteger extension

aschwaighofer commented on stdlib/public/core/IntegerParsing.swift:
```diff
@@ -83,17 +83,19 @@ where CodeUnits.Element : UnsignedInteger {
-}
-
 extension FixedWidthInteger {
+  /// _parseASCII function thunk that prevents inlining used as an
```
@dabrahams Like this?

dabrahams replied:
Better, thanks.

Other nits, should you care to address them:

1. This is not a doc comment; it's an implementation comment (a doc comment would begin with a sentence fragment describing what it _does_). Therefore it should not use triple slashes
2. I would prefer if you used something that described the role of this method, such as `SlowPath`, over something like `Outlined`, whose most common meaning is not at all what it's trying to communicate.

aschwaighofer commented:
@swift-ci Please benchmark

aschwaighofer commented:
@swift-ci Please smoke test

aschwaighofer committed:
Address review comments

Rename _parseASCIIOutlined to _parseASCIISlowPath.
Change doc comment to regular comment.

aschwaighofer commented:
@swift-ci Please smoke test
